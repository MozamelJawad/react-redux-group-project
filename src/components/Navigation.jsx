import React from 'react';
import { NavLink } from 'react-router-dom'; // Import Link and NavLink
import logo from '../assets/planet.png';

const Navigation = () => (
  <header>
    <div className="logo">
      <img src={logo} alt="Logo" />
      <span>Space Travelers&apos; Hub</span>
    </div>
    <nav>
      <ul>
        <li>
          <NavLink
            to="/"
            style={({ isActive }) => ({
              textDecoration: isActive ? 'underline' : 'none',
            })}
          >
            Rockets
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/missions"
            style={({ isActive }) => ({
              textDecoration: isActive ? 'underline' : 'none',
            })}
          >
            Missions
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/profile"
            style={({ isActive }) => ({
              textDecoration: isActive ? 'underline' : 'none',
            })}
          >
            My Profile
          </NavLink>
        </li>
      </ul>
    </nav>
  </header>
);

export default Navigation;
