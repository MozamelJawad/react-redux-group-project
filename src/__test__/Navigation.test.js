import React from 'react';
import { render } from '@testing-library/react';
import Navigation from '../components/Navigation';
import { BrowserRouter as Router} from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';


describe('Render header component correctly', () => {
    it('Should render the header correctly', () => {
        const { container } = render(
            <Router>
                <Navigation />
            </Router>,
        );
        expect(container).toMatchSnapshot();
    });

    it('should have an active link', () => {
        const {getByText} = render(
            <Router>
                <Navigation />
            </Router>,
        );

        const rocketsLink = getByText('Rockets');
        expect(rocketsLink).toBeInTheDocument();
    });
});
