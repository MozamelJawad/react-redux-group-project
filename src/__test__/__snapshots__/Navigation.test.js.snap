// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render header component correctly Should render the header correctly 1`] = `
<div>
  <header>
    <div
      class="logo"
    >
      <img
        alt="Logo"
        src="planet.png"
      />
      <span
        class="navbar"
      >
        Space Travelers' Hub
      </span>
    </div>
    <nav>
      <ul>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            style="text-decoration: underline;"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class=""
            href="/missions"
            style="text-decoration: none;"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            class=""
            href="/profile"
            style="text-decoration: none;"
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
